images/cfw/etc/init.d/S02backup_emmc

```diff
@@ -14,30 +30,27 @@ dump_emmc() {
         exit 0
     fi
 
-    # Get SN from cmdline
-    DEVICE_SN=$(cat /proc/cmdline | xargs -n1 | grep "bbl_serial=" | sed "s|bbl_serial=||")
-
     # Now that sd is mounted, do we need to do an emmc dump?
-    if [ -f /mnt/sdcard/x1plus/printers/$DEVICE_SN/dump-emmc ]; then
+    if x1p_get_setting boot.dump_emmc dump-emmc; then
         msg "Attempting to dump emmc as requested..." # Progress spinner sits ontop of this but that's fine
-        if [ -f /mnt/sdcard/x1plus/printers/$DEVICE_SN/emmc.bin ]; then
+        clear_flag # Clear now, so we unset even if there's no free disk space to do the dump
```

Comment:
...though maybe I would call it `_clear_flag`, since looking at this I go like "huh what is a bash command named clear_flag?".